src/Migrations/Utils/Functions.php

```diff
@@ -382,4 +381,90 @@ public static function create_media_text_block(
             $html
         );
     }
+
+    /**
+     * Create a new Embed block.
+     *
+     * @param string $social_media_url - The Social Media URL.
+     * @param string $type - The embed type.
+     * @param $provider - The embed provider.
+     *
+     * @return array - The new Embed block.
+     */
+
+    public static function create_embed_block(
+        string $social_media_url,
+        string $type,
+        string $provider,
+    ): array {
+        // phpcs:disable Generic.Files.LineLength.MaxExceeded
+        $html_content = '<figure class="wp-block-embed is-type-' . $type . ' is-provider-' . $provider . ' wp-block-embed-' . $provider . '"><div class="wp-block-embed__wrapper">
+        ' . $social_media_url . '
+        </div></figure>';
+        // phpcs:enable Generic.Files.LineLength.MaxExceeded
+
+        $attrs = [];
+        $attrs['type'] = $type;
+        $attrs['providerNameSlug'] = $provider;
+        $attrs['metadata']['name'] = $provider;
+        $attrs['url'] = $social_media_url;
+        $block['innerHTML'] = $html_content;
+        $block['innerContent'][0] = $html_content;
+
+        return self::create_new_block(
+            Constants::BLOCK_EMBED,
+            $attrs,
+            [],
+            $html_content,
+            [$html_content],
+        );
+    }
+
+    /**
+     * Create a new Group block.
+     *
+     * @param array $inner_blocks - The block's inner blocks.
+     * @param array $attrs - The block's attributes.
+     *
+     * @return array - The new Group block.
+     */
+
+    public static function create_group_block(array $inner_blocks, array $attrs): array
+    {
+        // IMPORTANT: DO NOT MODIFY THIS FORMAT!
+        $inner_html =
+        '<div class="wp-block-group">
+
+
+
+
+
+        </div>';
+
+        // IMPORTANT: DO NOT MODIFY THIS FORMAT!
+        $inner_content = array (
+            0 => '
+        <div class="wp-block-group">',
+            1 => null,
+            2 => '
+        ',
+            3 => null,
+            4 => '
+        ',
+            5 => null,
+            6 => '
+        ',
+            7 => null,
+            8 => '</div>
+        ',
+        );
```

Comment:
I really hate this (I know it is not your fault, actually I was the one who started implementing it this way). I just wonder if there is a cleaner way to build the blocks HTML.